src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -0,0 +1,1007 @@
+        if (op["arguments"]["session"]) {
+            auto session_name = op["arguments"]["session"].get_string().value.to_string();
+            auto& session = map.get_client_session(session_name);
+            return insert_one(map.get_collection(object), &session, op_view);
+        }
+        return insert_one(map.get_collection(object), nullptr, op_view);
+    }
+    if (name == "iterateUntilDocumentOrError")
+        return iterate_until_document_or_error(map.get_change_stream(object));
+    if (name == "failPoint")
+        return fail_point(map, apm, op_view);
+    if (name == "findOneAndUpdate")
+        return find_one_and_update(map.get_collection(object), op_view);
+    if (name == "listDatabases") {
+        map.get_client(object).list_databases().begin();
+        return empty_doc;
+    }
+    if (name == "assertSessionNotDirty") {
+        auto session_name = op["arguments"]["session"].get_string().value.to_string();
+        auto& session = map.get_client_session(session_name);
+        REQUIRE(!session.get_dirty());
+        return empty_doc;
+    }
+    if (name == "assertSessionDirty") {
+        auto session_name = op["arguments"]["session"].get_string().value.to_string();
+        auto& session = map.get_client_session(session_name);
+        REQUIRE(session.get_dirty());
+        return empty_doc;
+    }
+    if (name == "endSession") {
+        return end_session(map, object);
+    }
+    if (name == "assertSameLsidOnLastTwoCommands") {
+        auto cse1 = *(apm.end() - 1);
+        auto cse2 = *(apm.end() - 2);
+
+        CAPTURE(to_json(cse1), to_json(cse2));
+        REQUIRE(cse1["commandStartedEvent"]["command"]["lsid"].get_value() ==
+                cse2["commandStartedEvent"]["command"]["lsid"].get_value());
+        return empty_doc;
+    }
+    if (name == "assertDifferentLsidOnLastTwoCommands") {
+        auto cse1 = *(apm.end() - 1);
+        auto cse2 = *(apm.end() - 2);
+
+        CAPTURE(to_json(cse1), to_json(cse2));
+        REQUIRE(cse1["commandStartedEvent"]["command"]["lsid"].get_value() !=
+                cse2["commandStartedEvent"]["command"]["lsid"].get_value());
+        return empty_doc;
+    }
+    if (name == "startTransaction") {
+        auto& session = map.get_client_session(object);
+        return start_transaction(session, op_view);
+    }
+    if (name == "commitTransaction") {
+        auto& session = map.get_client_session(object);
+        session.commit_transaction();
+        return empty_doc;
+    }
+    if (name == "assertSessionTransactionState") {
+        auto session_name = op["arguments"]["session"].get_string().value.to_string();
+        auto& session = map.get_client_session(session_name);
+        return assert_session_transaction_state(session, op_view);
+    }
+    if (name == "dropCollection") {
+        auto coll_name = op["arguments"]["collection"].get_string().value.to_string();
+        auto& db = map.get_database(object);
+        db.collection(coll_name).drop();
```

Comment:
This should check if a session was passed.

Reply:
Done